### PR TITLE
fix(Popup): close previous popup on enter key

### LINKED ISSUE #866: Popup: previous popup is not closed on `Enter` key
## Bug Report

Previously opened `Popup` is not closed on `Enter` key click.

### Steps
1. Open https://stardust-ui.github.io/react/components/popup#variations-position
2. Click on any button
3. Press `Tab` and then `Enter` keys

### Expected Result
Only one popup is open.

### Actual Result
Two opened popups.

popup-enter

### Version
0.20.0

### PR DESCRIPTION
**Pull request by jongsue**
fixes #866 

I found the line that makes this bug

I tried to make the test case for this, but still failed it
even though it seems working fine
previous-popup-close

Popup's updateOutsideClickSubscription method has setTimeout
the test case runs before setTimeout callback

I hope someone can help me to make the test case pass

### CODE REVIEW

**Comment by layershifter** [comment]
Actually, we should use `EventStack` now to handle this, but it's not fully correct. I have a prototype that solves this issue (https://codesandbox.io/s/9yw3vq6774) and will be implemented in #949.

For now going to push changes to get it working.

File: packages/react/src/components/Popup/Popup.tsx [hunk 1]
```diff
@@ -251,9 +254,29 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
...
           },
         )
+        this.outsideKeySubscription = EventStack.subscribe(
+          'keydown',
+          e => {
+            const keyCode = keyboardKey.getCode(e)
+
+            const matchingKey = keyCode === keyboardKey.Enter || keyboardKey.Spacebar
+            const isOutsidePopupElement =
+              this.popupDomElement && !this.popupDomElement.contains(e.target)
```
Comment (bmdalex) [inline packages/react/src/components/Popup/Popup.tsx]: can you extract this logic from `updateOutsideClickSubscription` so we can reuse here?

**Review by bmdalex — APPROVED** [review]
thanks for fixing 👍 one comment
fyi @kuzhelov

**Comment by jongsue** [comment]
I did extra method to reuse it as @Bugaa92 said

Thank you guys for the reviews
I've learned a lot of good points here

**Comment by kuzhelov** [comment]
please, just update `Fixes` section of changelog before merge 👍

**Pull request merged**